kernel/src/hil/rng.rs

```diff
@@ -1,9 +1,10 @@
-//! Interfaces for accessing a random number generator.
+//! Interfaces for random number generation.
 //!
 //! A random number generator produces a stream of random numbers, either from
 //! hardware or based on an initial seed. The [RNG](trait.RNG.html) trait
 //! provides a simple, implementation agnostic interface for getting new random
-//! values.
+//! values. Cryptosystems should use this interface with care, as it does not
+//! provide strong entropy guarantees.
```

Comment:
Should we consider renaming this interface to make this more explicit? There's been a bit of a backlash in other communities where historically there were `random` and `secure_random` APIs, and many folks understandably mistakenly use the former. Perhaps we have a chance to do better :)

I'm not sure what the better than here is, I don't think it's exactly `insecure_rng` or `low_entropy_rng` or `no_promises_rng`, as those are too negative. Maybe we should try to define the strong entropy guarantees that we would imagine a complimentary interface providing and use the delta to think about this further?

Reply:
One of the big issues here is the tradeoffs of different kinds of randomness. If you are not using the random functions for cryptographic operations (e.g., for random backoff), entropy guarantees aren't really needed.

If we want to be very careful about this, we could use types such that only high-entropy randomness is fed to cryptographic operations.

Reply:
Using types to enforce high entropy for crypto operations is really cool! (Albeit somewhat orthogonal to this PR)

Should this interface return `MediumEntropyBytes` that implement `to` for `u32`, or is that just needlessly pedantic?